ladybug/_datacollectionbase.py

```diff
@@ -407,6 +414,74 @@ def are_collections_aligned(data_collections, raise_exception=True):
                     return False
         return True
 
+    @staticmethod
+    def compute_function_aligned(funct, data_collections, result_type, result_unit):
+        """Compute a function with a list of aligned data collections or individual values.
+
+        Args:
+            funct: A function with a single numerical value as output and one or
+                more numerical values as input.
+            data_collections: A list with a length equal to the number of arguments
+                for the function. Items of the list can be either Data Collections
+                or individual values to be used at each datetime of other collections.
+            result_type: An instance of a Ladybug data type that describes the resulting
+                numerical values.
+            result_unit: The units of the resulting numerical values.
+
+        Return:
+            A Data Collection with the results function. If all items in this list of
+            data_collections are individual values, only a single value will be returned.
+
+        Usage:
+            from ladybug.datacollection import HourlyContinuousCollection
+            from ladybug.epw import EPW
+            from ladybug.psychrometrics import humid_ratio_from_db_rh
+            from ladybug.datatype.percentage import HumidityRatio
+
+            epw_file_path = './epws/denver.epw'
+            denver_epw = EPW(epw_file_path)
+            pressure_at_denver = 85000
+            hr_inputs = [denver_epw.dry_bulb_temperature,
+                         denver_epw.relative_humidity,
+                         pressure_at_denver]
+            humid_ratio = HourlyContinuousCollection.compute_function_aligned(
+                humid_ratio_from_db_rh, hr_inputs, HumidityRatio(), 'fraction')
+            # humid_ratio will be a Data Colleciton of humidity ratios at Denver
+        """
+        # first check the input collections or individual values
+        data_c = []
+        numerical_vals = []
+        for func_input in data_collections:
+            if isinstance(func_input, BaseCollection):
+                data_c.append(func_input)
+            else:
+                try:
+                    numerical_vals.append(float(func_input))
+                except ValueError:
+                    raise TypeError('Expected a number or a Data Colleciton. '
+                                    'Got {}'.format(type(func_input)))
+
+        # run the function and return the result
+        if len(data_c) == 0:
+            # just inidivdual numerical values
+            return funct(*numerical_vals)
+        else:
+            # data collections
+            aligned_collections = []
+            for func_input in data_collections:
```

Comment:
Can't all this be done when you are doing the initial check in line 454?

It' not critical but performance-wise you will go through all the data once.

Reply:
You were definitely right to point out that this could have been written a lot more elegantly and without the need to create a bunch of new lists (or new aligned data collections).  So I have cleaned it up a lot in the commit that I am about to push.  I found that there is a way to do it without any double iteration but it involves doing an `if` statement inside of a nested `for` loop like so:

`           result = data_colls[0].get_aligned_collection(data_type=data_type, unit=unit)
            for i in xrange(len(data_colls[0].values)):
                funct_input = []
                for col in data_collections:
                    funct_input.append(col if isinstance(col, float) else col[i])
                result[i] = funct(*funct_input)
            return result
`

I was curious if it would actually be faster so I profiled it and it turns out that the double iteration was a bit faster.  Granted, we are only talking about a difference of ~10ms in an operation that takes 40ms to run so it is nothing major.  But, given that it's basically the same number of lines of code and both of them seem to be equally readable, I am sticking with the double iteration.